### PR TITLE
Feature lwdev 7722 instruments advanced search

### PR DESCRIPTION
**Pull request by GeorgePuisha**
https://lykkex.atlassian.net/browse/LWDEV-7722

### CODE REVIEW

File: src/stores/__tests__/referenceStore.spec.ts [hunk 1]
```diff
@@ -111,6 +129,44 @@ describe('referenceStore', () => {
...
+        Promise.resolve({
+          Id: '1',
+          FullName: 'Lykke'
+        })
+      );
+
+      await assetStore.fetchAssetById('1');
+
+      expect(assetStore.getAssets()).not.toBeNull();
+      expect(assetStore.getAssets().length).toBe(1);
```
Comment (PaulRumyantsevItr) [inline src/stores/__tests__/referenceStore.spec.ts]: .toHaveLength(1)

File: src/stores/__tests__/referenceStore.spec.ts [hunk 2]
```diff
@@ -278,4 +356,50 @@ describe('referenceStore', () => {
...
+          price: 1000,
+          bid: 1,
+          ask: 1,
+          volumeInBase: 0,
+          change24h: 0,
+          volume: 0
+        })
+      );
+
+      expect(refStore.findInstruments('BTC', '').length).not.toEqual(0);
```
Comment (PaulRumyantsevItr) [inline src/stores/__tests__/referenceStore.spec.ts]: Same, .toHaveLength

File: src/stores/referenceStore.ts [hunk 3]
```diff
@@ -78,8 +78,10 @@ class ReferenceStore extends BaseStore {
 
   findInstruments = (term: string, watchlistName: string) => {
     const {getWatchlistByName} = this.rootStore.watchlistStore;
-    const instrumentsByName = this.instruments.filter(instrument =>
-      includes(normalize(term), normalize(instrument.displayName!))
+    const instrumentsByName = this.instruments.filter(
+      instrument =>
+        includes(normalize(term), normalize(instrument.displayName!)) ||
```
Comment (PaulRumyantsevItr) [inline src/stores/referenceStore.ts]: Move result of ```normalize(term)``` to separate variable

File: src/stores/referenceStore.ts [hunk 4]
```diff
@@ -115,32 +117,52 @@ class ReferenceStore extends BaseStore {
...
-            this.assets = dtoAssets.map((x: any) =>
-              mappers.mapToAsset(x, this.categories)
-            );
+    const requests = [this.api.fetchAll(), this.api.fetchAssetsDescriptions()];
+
+    return Promise.all(requests).then(data => {
+      const assets = data[0];
+      const descriptions = data[1];
+      if (
+        assets &&
```
Comment (PaulRumyantsevItr) [inline src/stores/referenceStore.ts]: assets && assets.Assets &&
descriptions && descriptions.Descriptions

File: src/stores/referenceStore.ts [hunk 5]
```diff
@@ -115,32 +117,52 @@ class ReferenceStore extends BaseStore {
...
+        descriptions &&
+        assets.Assets &&
+        descriptions.Descriptions
+      ) {
+        runInAction(() => {
+          this.assets = assets.Assets.map((asset: any) => {
```
Comment (PaulRumyantsevItr) [inline src/stores/referenceStore.ts]: Try to use ```any``` only if you can't find a type. Assets has types

File: src/stores/referenceStore.ts [hunk 6]
```diff
@@ -115,32 +117,52 @@ class ReferenceStore extends BaseStore {
...
+            const description =
+              descriptions.Descriptions.find(
+                (desc: any) => desc.Id === asset.Id
```
Comment (PaulRumyantsevItr) [inline src/stores/referenceStore.ts]: Same for desc (and rename to ```description``` too please)

File: src/stores/referenceStore.ts [hunk 7]
```diff
@@ -115,32 +117,52 @@ class ReferenceStore extends BaseStore {
...
-      return Promise.resolve(asset);
+    const requests = [
+      this.api.fetchAssetById(id),
+      this.api.fetchAssetDescriptionById(id)
+    ];
+
+    return Promise.all(requests).then(data => {
+      const asset = data[0];
+      const description = data[1];
+      let mappedAsset;
```
Comment (PaulRumyantsevItr) [inline src/stores/referenceStore.ts]: Maybe it make sense to set null as default value?

File: src/api/assetApi.ts [hunk 8]
```diff
@@ -23,6 +23,9 @@ export class RestAssetApi extends RestApi implements AssetApi {
   fetchAssetById = (id: string) => this.get(`/assets/${id}`);
   fetchRates = () => this.get('/assetpairs/rates');
   fetchMarket = () => this.get('/markets');
+  fetchAssetsDescriptions = () => this.get('/assets/description');
```
[the remark this answers is not in the document]
Comment (GeorgePuisha) [reply src/api/assetApi.ts]: An array is returned, that's why `s` takes place.

File: src/stores/referenceStore.ts (hunk 4 above)
Comment (sergeyschegolev) [inline src/stores/referenceStore.ts]: You should delete assets.Assets verification

**Pull request closed without merging**